spring-integration-redis/src/test/java/org/springframework/integration/redis/util/RedisLockRegistryTests.java

```diff
@@ -427,6 +429,33 @@ void testExceptionOnExpire(RedisLockType testRedisLockType) throws Exception {
 		registry.destroy();
 	}
 
+	@ParameterizedTest
+	@EnumSource(RedisLockType.class)
+	void testRenewalOnExpire(RedisLockType redisLockType) throws Exception {
+		RedisLockRegistry registry = new RedisLockRegistry(redisConnectionFactory, this.registryKey, 3_000L);
+		registry.setRenewalScheduledExecutor(new ScheduledThreadPoolExecutor(10, new CustomizableThreadFactory("renewal-schedule-thread")));
+		registry.setRedisLockType(redisLockType);
+		Lock lock1 = registry.obtain("foo");
+		assertThat(lock1.tryLock()).isTrue();
+		waitForExpire2("foo");
```

Comment:
I think this test has to be modeled with a shorter `expireAfter` and instead of Redis call, we indeed should just `Thread.sleep(expireAfter * 2)` and then verify that lock was not expired because we have that renewal task on background.